zap/src/main/java/org/zaproxy/zap/extension/alert/AlertTreeModel.java

```diff
@@ -247,6 +259,23 @@ public synchronized void deletePath(Alert alert) {
         }
     }
 
+    public boolean isSystemicAlert(Alert alert, boolean countAlert) {
+        if (!alert.isSystemic()) {
+            return false;
+        }
+        try {
+            Map<String, AtomicInteger> m =
+                    siteToSystemicAlertMap.computeIfAbsent(
+                            SessionStructure.getHostName(alert.getMsgUri()),
+                            a -> new ConcurrentHashMap<>());
+            AtomicInteger ai = m.computeIfAbsent(alert.getAlertRef(), a -> new AtomicInteger());
+            return ExtensionAlert.isOverSystemicLimit(countAlert ? ai.incrementAndGet() : ai.get());
+        } catch (URIException e) {
+            // Ignore
+        }
+        return false;
+    }
```

Comment:
Fail fast and reduce lambda nesting.

```java
public boolean isSystemicAlert(Alert alert, boolean countAlert) {
    if (!alert.isSystemic()) {
        return false;
    }

    final String host;
    try {
        host = SessionStructure.getHostName(alert.getMsgUri());
    } catch (URIException e) {
        return false;
    }

    Map<String, AtomicInteger> alertMap = siteToSystemicAlertMap.computeIfAbsent(
            host, k -> new ConcurrentHashMap<>());

    AtomicInteger counter = alertMap.computeIfAbsent(alert.getAlertRef(), k -> new AtomicInteger());

    int currentCount = countAlert ? counter.incrementAndGet() : counter.get();
    return ExtensionAlert.isOverSystemicLimit(currentCount);
}
```

Reply:
The URIException should never happen.
I dont see how this code is any better tbh - is this for readability or ??

Reply:
Mainly it makes it clear what's actually being "tried", I agree it should essentially never happen, but it is actually on the one line not the rest of the handling.

I'm not stuck on it, I'm happy to approve either way.